pkg/target/cluster.go

```diff
@@ -47,6 +52,19 @@ func (cl *Cluster) resolveHosts(path string) ([]*Host, error) {
 		return []*Host{
 			cl.Hosts[jumpHash(path, len(cl.Hosts))],
 		}, nil
+	case conf.LB:
+		cl.Hm.RLock()
+		defer cl.Hm.RUnlock()
+		availableHostCount := len(cl.AvailableHosts)
+		key := fnv1a.HashString64(path)
+		if availableHostCount == 0 {
```

Comment:
In the future, we could make this configurable as percentage of len(Hosts).
Idea is that we if we have 10 hosts and only one is available we still don't want to send only to it as it would not have enough capacity anyway.

Reply:
and what happens if we don't do this? we will get host throttling, which is ok. so, it's a problem that handles itself.

Reply:
@avereha agreed. @grzkv If we don't, we risk bringing down the few central relays still available by overloading them

Reply:
Correct. We could add this later so we don't overcomplicate current PR.